app/src/main/kotlin/com/wire/android/mapper/MessageMapper.kt

```diff
@@ -50,8 +50,32 @@ class MessageMapper @Inject constructor(
             userList = userList
         )
 
-        val footer = if(message is Message.Regular) {
-            MessageFooter(message.id, message.reactions.totalReactions, message.reactions.selfUserReactions)
+        val footer = if (message is Message.Regular) {
+            // TODO find ugly and proper heart emoji and merge them to ugly one 😅
+            val totalHeartsCount = message.reactions.totalReactions
+                .filterKeys { isHeart(it) }.values
+                .scan(0) { acc, count -> acc + count }
+                .lastOrNull()
```

Comment:
Kotlin's stdlib has a [sumOf](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/sum-of.html) and for collections of numbers, [sum](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/sum.html) also exists to make things simpler.

```suggestion
                .sum()
```